cmd/ipfs/main.go

```diff
@@ -180,6 +181,8 @@ func (i *cmdInvocation) Run(ctx context.Context) (output io.Reader, err error) {
 		return nil, err
 	}
 
+	interrupt.Wait()
```

Comment:
yeah it would be easier to reason about things if we didnt have to know this is a waitgroup:

```
interrupt.Close()
```